api/files/serializers.py

```diff
@@ -421,15 +427,20 @@ def create(self, validated_data):
 
 class FileDetailSerializer(FileSerializer):
     """
-    Overrides FileSerializer to make id required.
+    - Overrides FileSerializer to make id required
+    - Files should return the id type they are queried with, but only in osfstorage is the id is reliably equivalent to
+     the path attribute, so that should not be overridden.
+
     """
-    id = IDField(source='_id', required=True)
+    id = GuidOrIDField(source='_id', required=True)
 
     def to_representation(self, value):
         data = super().to_representation(value)
-        guid = Guid.load(self.context['view'].kwargs['file_id'])
+        view = self.context['view']
+        guid = Guid.load(view.kwargs['file_id'])
         if guid:
             data['data']['id'] = guid._id
+            data['data']['links']['self'] = absolute_reverse(f'{view.view_category}:{view.view_name}', kwargs=view.kwargs)
```

Comment:
what happens if we just always update the self link? Shouldn't it "Just Work"©. A possible meaningless update (with an informative comment) feels like a better decision than an unnecessary database lookup (as we're not using that guid value here at all)

Reply:
good point, I'll change it to avoid the lookup.